src/vs/base/common/comparers.ts

```diff
@@ -6,30 +6,168 @@
 import { sep } from 'vs/base/common/path';
 import { IdleValue } from 'vs/base/common/async';
 
-const intlFileNameCollator: IdleValue<{ collator: Intl.Collator, collatorIsNumeric: boolean }> = new IdleValue(() => {
-	const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'base' });
+// When comparing large numbers of strings, such as in sorting large arrays, is better for
+// performance to create an Intl.Collator object and use the function provided by its compare
+// property than it is to use String.prototype.localeCompare()
+
+// A collator with numeric sorting enabled.
+const intlFileNameCollatorNumeric: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: true });
+	return {
+		collator: collator
+	};
+});
+
+// A collator with numeric sorting enabled, and sensitivity to accents but not case.
+const intlFileNameCollatorNumericCaseInsenstive: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'accent' });
 	return {
-		collator: collator,
-		collatorIsNumeric: collator.resolvedOptions().numeric
+		collator: collator
 	};
 });
 
-export function compareFileNames(one: string | null, other: string | null, caseSensitive = false): number {
-	const a = one || '';
-	const b = other || '';
-	const result = intlFileNameCollator.getValue().collator.compare(a, b);
+// A collator with numeric sorting disabled.
+const intlFileNameCollator: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: false });
+	return {
+		collator: collator
+	};
+});
 
-	// Using the numeric option in the collator will
-	// make compare(`foo1`, `foo01`) === 0. We must disambiguate.
-	if (intlFileNameCollator.getValue().collatorIsNumeric && result === 0 && a !== b) {
-		return a < b ? -1 : 1;
+// A collabor with numeric sorting disabled and sensitivity to accents but not case.
+const intlFileNameCollatorCaseInsensitive: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: false, sensitivity: 'accent' });
+	return {
+		collator: collator
+	};
+});
+
+// Export compareFileNamesNumeric by its old name too for backwards compatibility.
```

Comment:
I wasn't sure what to do about the reference to compareFileNames in src/vs/workbench/test/browser/api/extHostDocumentData.test.ts so this is what I wound up doing.  

@isidorn can you advise? 

Would it be okay to delete this extra export? And replace or delete the reference in extHostDocumentData? What about the placeholder parameter? 

No code is currently using that parameter (and it was being ignored before too), but I wasn't sure if the fact that this function is listed in extHostDocumentData - even if just in the test code - means it is part of some sort of API which needs to be kept compatible? 

Also editing that data directly seemed like it might not be a good idea since it looked like it was generated using some sort of script.

Reply:
Hmm... it seems that tree shaking mangles this extra export, which is causing the failed check in CI. Apparently here's what this part of the `comparers.ts` file looks like after tree-shaking - which is definitely not right.

```ts
// A collator with numeric sorting enabled, and sensitivity to accents but not case.
const intlFileNameCollatorNumericCaseInsenstive: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
	const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'accent' });
	return {
		collator: collator
	};
}); compareFileNamesNumeric as compareFileNames
```

It also seems to remove all of my new functions - which is probably the real problem here. They are used in other parts of the code? What's the best way to fix this?

Reply:
Resolved the tree shaking issue. Instead of exporting `compareFileNamesNumeric` as both `compareFileNames` and `compareFileNamesNumeric`, I added a `compareFileNames` function that calls `compareFileNamesNumeric`.